File: MG/App/Backend/mojGradApp/mojGradApp/Models/InfoModels/UserStatisticInfo.cs
using Org.BouncyCastle.Asn1.Mozilla;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace mojGradApp.Models.InfoModels
{
    public class UserStatisticInfo
    {
        public int NumberOfCitizens { get; set; }
        public int NumberOfInstitutions { get; set; }
        public int CityID { get; set; }
        public String CityName { get; set; }
        public DateTime Time { get; set; }
        public int Day { get; set; }
        public int Month { get; set; }
        public int Year { get; set; }
        public int TotalNumber { get; set; }

        public UserStatisticInfo() { }

        public UserStatisticInfo(UserStatistics us, bool withCity) 
        {
            NumberOfCitizens = us.NumberOfCitizens;
            NumberOfInstitutions = us.NumberOfInstitutions;
            if(withCity == true)
            {
                CityID = us.CityID;
                CityName = us.City.Name;
            }
            Time = us.Time;
            Day = us.Time.Day;
            Month = us.Time.Month;
            Year = us.Time.Year;
            TotalNumber = us.NumberOfCitizens + us.NumberOfInstitutions;
        }
    }
}

File: MG/App/Frontend/WEB_flutter/lib/models/modelsViews/userStatisticInfo.dart
class UserStatisticInfo
{
  int numberOfCitizens;
  int numberOfInstitutions;
  int cityID;
  String cityName;
  DateTime time;
  int day;
  int month;
  int year;
  int totalNumber;

  UserStatisticInfo(numberOfCitizens, numberOfInstitutions, cityID, cityName, time, day, month, year, totalNumber)
  {
    this.numberOfCitizens = numberOfCitizens;
	this.numberOfInstitutions = numberOfInstitutions;
	this.cityID = cityID;
	this.cityName = cityName;
	this.time = time;
	this.day = day;
	this.month = month;
	this.year = year;
	this.totalNumber = totalNumber;
  }

  Map<String, dynamic> toMap() 
  {
    var map = Map<String, dynamic>();
    map['numberOfCitizens'] = numberOfCitizens;
    map['numberOfInstitutions'] = numberOfInstitutions;
	map['cityID'] = cityID;
	map['cityName'] = cityName;
	map['time'] = time.toIso8601String();
	map['day'] = day;
	map['month'] = month;
	map['year'] = year;
	map['totalNumber'] = totalNumber;
    return map;
  }

  UserStatisticInfo.fromObject(dynamic o) 
  {
    this.numberOfCitizens = o['numberOfCitizens'];
    this.numberOfInstitutions = o['numberOfInstitutions'];
	this.cityID = o['cityID'];
	this.cityName = o['cityName'];
	this.time = DateTime.tryParse(o['time']);
	this.day = o['day'];
	this.month = o['month'];
	this.year = o['year'];
	this.totalNumber = o['totalNumber'];
  }
  
}